example_error_handler_test.go

```diff
@@ -0,0 +1,113 @@
+package river_test
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/riverqueue/river"
+	"github.com/riverqueue/river/internal/riverinternaltest"
+	"github.com/riverqueue/river/internal/util/slogutil"
+	"github.com/riverqueue/river/riverdriver/riverpgxv5"
+)
+
+type CustomErrorHandler struct{}
+
+func (*CustomErrorHandler) HandleError(ctx context.Context, job *river.JobRow, err error) *river.ErrorHandlerResult {
+	fmt.Printf("Job errored with: %s\n", err)
+
+	// Either function can also set the job to be immediately cancelled, which
+	// we take advantage of here to make sure it's not retried in the example.
+	// Can also be `return nil`.
+	return &river.ErrorHandlerResult{SetCancelled: true}
```

Comment:
See note here, but sort of had to set cancelled because otherwise there's a chance the job could retry before the example finishes up.

I guess I could also set max attempts to 1. Not sure which one is better.

Reply:
I'm torn. It's useful for this example to also demonstrate `SetCancelled` because there's nowhere else we'd do that. However I worry a bit about people copy-pasta'ing this without realizing what they're doing, particularly in `HandlePanic` where there's no comment about it.

Maybe the answer is to do both? If we set MaxAttempts to 1 for the entire worker, then we could easily show off both the `SetCancelled: true` usage as well as the more common usage where you probably don't want to cancel any job that panics or errors.

If we go that route, I'd say it makes the most sense to put the `SetCancelled` in `HandlePanic` so that `HandleError` can demonstrate the most common usage.

Reply:
+1 all that. Changed to max attempts of 1, and made `HandleError` return nil for copy/pasta convenience.